packages/devtools_app/lib/src/memory/memory_protocol.dart

```diff
@@ -144,26 +140,74 @@ class MemoryTracker {
     _recalculate(true);
   }
 
-  // Poll ADB meminfo
+  /// Poll ADB meminfo
   Future<AdbMemoryInfo> _fetchAdbInfo() async =>
       AdbMemoryInfo.fromJson((await serviceManager.getAdbMemoryInfo()).json);
 
+  /// Returns the MemoryUsage of a particular isolate.
+  /// @param id isolateId.
+  /// @param usage associated with the passed in isolate's id.
+  /// @returns the MemoryUsage of the isolate or null if isolate is a sentinal.
+  Future<MemoryUsage> _isolateMemoryUsage(String id, MemoryUsage usage) async =>
+      await memoryController.isIsolateLive(id) ? usage : null;
+
   void _recalculate([bool fromGC = false]) async {
     int used = 0;
     int capacity = 0;
     int external = 0;
 
-    for (MemoryUsage memoryUsage in isolateHeaps.values) {
-      used += memoryUsage.heapUsage;
-      capacity += memoryUsage.heapCapacity;
-      external += memoryUsage.externalUsage;
+    final keysToRemove = <String>[];
+
+    final isolateCount = isolateHeaps.length;
+    final keys = isolateHeaps.keys.toList();
+    for (var index = 0; index < isolateCount; index++) {
+      final isolateId = keys[index];
+      var usage = isolateHeaps[isolateId];
+      if (isolateCount > 1) {
+        // Is the isolate live (not sentinaled).
+        usage = await _isolateMemoryUsage(isolateId, usage);
```

Comment:
make this a different variable rather than reusing the usage object. 
this current logic seems a bit suspect or I am getting confused. it looks like we are assuming that if there is one isolate it can't be a sentinel which seems strange.

Reply:
Thought I was being cute only checking if isolates > 1 however, should always do then in case something strange from the normal mechanism happens we'll still work.